storage/rawstores.go

```diff
@@ -54,8 +56,51 @@ func createHTTPClient(cfg HTTPClientConfig) *http.Client {
+			CacheHit:        scope.MustNewCounter("cache_hit", "Number of times metadata was found in cache"),
+			CacheMiss:       scope.MustNewCounter("cache_miss", "Number of times metadata was not found in cache and remote fetch was required"),
+			CacheWriteError: scope.MustNewCounter("cache_write_err", "Failed to write to cache"),
+		},
+		protoMetrics: &protoMetrics{
+			FetchLatency:                 scope.MustNewStopWatch("proto_fetch", "Time to read data before unmarshalling", time.Millisecond),
+			MarshalTime:                  scope.MustNewStopWatch("marshal", "Time incurred in marshalling data before writing", time.Millisecond),
+			UnmarshalTime:                scope.MustNewStopWatch("unmarshal", "Time incurred in unmarshalling received data", time.Millisecond),
+			MarshalFailure:               scope.MustNewCounter("marshal_failure", "Failures when marshalling"),
+			UnmarshalFailure:             scope.MustNewCounter("unmarshal_failure", "Failures when unmarshalling"),
+			WriteFailureUnrelatedToCache: scope.MustNewCounter("write_failure_unrelated_to_cache", "Raw store write failures that are not caused by ErrFailedToWriteCache"),
+			ReadFailureUnrelatedToCache:  scope.MustNewCounter("read_failure_unrelated_to_cache", "Raw store read failures that are not caused by ErrFailedToWriteCache"),
+		},
+		copyMetrics: newCopyMetrics(scope.NewSubScope("copy")),
+		stowMetrics: &stowMetrics{
+			BadReference: labeled.NewCounter("bad_key", "Indicates the provided storage reference/key is incorrectly formatted", scope, labeled.EmitUnlabeledMetric),
+			BadContainer: labeled.NewCounter("bad_container", "Indicates request for a container that has not been initialized", scope, labeled.EmitUnlabeledMetric),
+
+			HeadFailure: labeled.NewCounter("head_failure", "Indicates failure in HEAD for a given reference", scope, labeled.EmitUnlabeledMetric),
+			HeadLatency: labeled.NewStopWatch("head", "Indicates time to fetch metadata using the Head API", time.Millisecond, scope, labeled.EmitUnlabeledMetric),
+
+			ReadFailure:     labeled.NewCounter("read_failure", "Indicates failure in GET for a given reference", scope, labeled.EmitUnlabeledMetric, failureTypeOption),
+			ReadOpenLatency: labeled.NewStopWatch("read_open", "Indicates time to first byte when reading", time.Millisecond, scope, labeled.EmitUnlabeledMetric),
+
+			WriteFailure: labeled.NewCounter("write_failure", "Indicates failure in storing/PUT for a given reference", scope, labeled.EmitUnlabeledMetric, failureTypeOption),
+			WriteLatency: labeled.NewStopWatch("write", "Time to write an object irrespective of size", time.Millisecond, scope, labeled.EmitUnlabeledMetric),
+		},
+	}
+}
+
 // NewDataStore creates a new Data Store with the supplied config.
-func NewDataStore(cfg *Config, metricsScope promutils.Scope) (s *DataStore, err error) {
+func NewDataStore(cfg *Config, metrics *DataStoreMetrics) (s *DataStore, err error) {
```

Comment:
let's keep the constructor as is, but also add a method `func (d *DataStore) RefreshConfig(cfg *Config) error` that can re-use code defined in this constructor (this shared code can be pulled out to a separate private func for re-use)